app/utils/sslyze/grade_scan_result.py

```diff
@@ -21,56 +23,129 @@ class Grades(Enum):
-
-        if self.partial_simplified.tlsv13_working_weak_ciphers_count == 0 and \
-                self.partial_simplified.tlsv12_working_weak_ciphers_count == 0:
-            self._format_msg_and_cap(Grades.B, "doesn't support either TLS 1.2 or TLS 1.3")
-
-    def get_result(self) -> Tuple[str, List[str]]:
-        self._calculate()
-        return self.grade_cap.name, self.grade_cap_reasons
+        if self.partial_simplified.tlsv12_working_ciphers_count:
+            self._format_msg_and_cap(Grades.A, "supports TLS 1.2")
+        if self.partial_simplified.tlsv13_working_ciphers_count:
+            self._format_msg_and_cap(Grades.A_plus, "supports TLS 1.3")
+
+    def calculate_headers(self):
+        sec_header: db_models.HTTPSecurityHeaders = self.scan_result.http_security_headers
+        if sec_header is None:
+            logger.error("SC0013 ScanResult has no http security headers object associated")
+            return
+        if sec_header.strict_transport_security_header and \
+                sec_header.public_key_pins_header:
+            self._format_msg_and_cap(Grades.A_plus, "supports HKPK and HSPS")
+        elif sec_header.strict_transport_security_header:
+            self._format_msg_and_cap(Grades.A, "supports HSPS")
+
+    def calculate_renegotiation(self):
+        renegotiation = self.scan_result.session_renegotiation
+        if renegotiation is None:
+            logger.error("SC0014 ScanResult has no renegotiation object associated")
+            return
+        if renegotiation.supports_secure_renegotiation:
+            self._format_msg_and_cap(Grades.A_plus, "supports secure client renegotiation")
+        else:
+            self._format_msg_and_cap(Grades.F, "supports client renegotiation, but is not secure")
+
+    def calculate_cipher_strength(self):
+        if self.partial_simplified.tlsv13_working_weak_ciphers_count is not None:
+            self._format_msg_and_cap(Grades.A, "has TLS 1.3 with weak ciphers")
+        if self.partial_simplified.tlsv12_working_weak_ciphers_count is not None:
+            self._format_msg_and_cap(Grades.A, "has TLS 1.2 with weak ciphers")
+        elif self.partial_simplified.tlsv11_working_weak_ciphers_count is not None:
+            self._format_msg_and_cap(Grades.B, "has TLS 1.1 with weak ciphers")
+
+    def calculate_vulnerabilities(self):
+        if self.scan_result.openssl_ccs_injection is not None and self.scan_result.openssl_ccs_injection.is_vulnerable_to_ccs_injection:
+            self._format_msg_and_cap(Grades.F, "Vulnerable to CSS injection")
+        if self.scan_result.openssl_heartbleed is not None and self.scan_result.openssl_heartbleed.is_vulnerable_to_heartbleed:
+            self._format_msg_and_cap(Grades.F, "Vulnerable to heartbleed")
+        if self.scan_result.downgrade_attacks is not None and not self.scan_result.downgrade_attacks.supports_fallback_scsv:
+            self._format_msg_and_cap(Grades.F, "Vulnerable to downgrade attacks")
```

Comment:
Add empty line to end of file
```suggestion
            self._format_msg_and_cap(Grades.F, "Vulnerable to downgrade attacks")

```